lib/sles4sap/sap_deployment_automation_framework/configure_tfvars.pm

```diff
@@ -0,0 +1,121 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::sap_deployment_automation_framework::configure_tfvars;
+
+use strict;
+use warnings;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use utils qw(file_content_replace);
+use sles4sap::azure_cli qw(az_storage_blob_list);
+use sles4sap::sap_deployment_automation_framework::deployment qw(get_os_variable);
+use sles4sap::sap_deployment_automation_framework::deployment_connector qw(find_deployment_id);
+
+=head1 SYNOPSIS
+
+Library with common functions for Microsoft SDAF deployment automation that help with preparation of tfvars file.
+
+=cut
+
+our @EXPORT = qw(
+  prepare_tfvars_file
+);
+
+=head2 prepare_tfvars_file
+
+    prepare_tfvars_file(deployment_type=>$deployment_type);
+
+=over 1
+
+=item B<$deployment_type>: Type of the deployment (workload_zone, sap_system, library... etc)
+
+=back
+
+Downloads tfvars template files from openQA data dir and places them into correct place within SDAF repo structure.
+Returns full path of the tfvars file.
+
+=cut
+
+sub prepare_tfvars_file {
+    my (%args) = @_;
+    croak 'Deployment type not specified' unless $args{deployment_type};
+    my %tfvars_os_variable = (
+        deployer => 'deployer_parameter_file',
+        sap_system => 'sap_system_parameter_file',
+        workload_zone => 'workload_zone_parameter_file',
+        library => 'library_parameter_file'
+    );
+    croak "Unknown deployment type: $args{deployment_type}" unless $tfvars_os_variable{$args{deployment_type}};
+
+    my %tfvars_template_url = (
+        deployer => data_url('sles4sap/sap_deployment_automation_framework/DEPLOYER.tfvars'),
+        sap_system => data_url('sles4sap/sap_deployment_automation_framework/SAP_SYSTEM.tfvars'),
+        workload_zone => data_url('sles4sap/sap_deployment_automation_framework/WORKLOAD_ZONE.tfvars'),
+        library => data_url('sles4sap/sap_deployment_automation_framework/LIBRARY.tfvars')
+    );
+
+    # replace default vnet name with shorter one to avoid naming restrictions
+    set_workload_vnet_name();
+
+    my $tfvars_file = get_os_variable($tfvars_os_variable{$args{deployment_type}});
+    my $retrieve_tfvars_cmd = join(' ', 'curl', '-v', '-fL', $tfvars_template_url{$args{deployment_type}}, '-o', $tfvars_file);
+
+    assert_script_run($retrieve_tfvars_cmd);
```

Comment:
Why not do directly:
```
assert_script_run("curl -v -fL $tfvars_template_url{$args{deployment_name}} -o $tfvars_file");
```
The line ends up shorter and you avoid creating a new variable.

Reply:
True, makes sense. 
Fixed.